### PR TITLE
feat: Handle stale rate in chainlink oracle

### PR DESCRIPTION
**Pull request by fzavalia**

### CODE REVIEW

File: contracts/mocks/DummyDataFeed.sol [hunk 1]
```diff
@@ -11,14 +11,17 @@ import '../interfaces/IDataFeed.sol';
 contract DummyDataFeed is IDataFeed {
     uint8 stateDecimals;
     int256 answer;
+    uint256 answeredAtOffset;
 
     /**
      * @param _decimals - Number of decimals the decimals function has to return
      * @param _answer - Value returned as the second argument of the return value of both getRoundData and latestRoundData
+     * @param _answeredAtOffset - Value deducted from the fourth argument of the return value of both getRoundData and latestRoundData which is the current block timestamp
```
Comment (nachomazzara) [inline contracts/mocks/DummyDataFeed.sol:19]: Im not sure if this is the right description. It is not just the offset?

File: contracts/oracles/ChainlinkOracle.sol [hunk 2]
```diff
@@ -31,7 +34,9 @@ contract ChainlinkOracle is IOracle {
     function getRate() external view override returns (uint256) {
         uint256 feedDecimals = uint256(dataFeed.decimals());
 
-        (, int256 rate, , , ) = dataFeed.latestRoundData();
+        (, int256 rate, , uint256 updatedAt, ) = dataFeed.latestRoundData();
+        
+        require(updatedAt >= block.timestamp - tolerance, "ChainlinkOracle#getRate: STALE_RATE");
```
Comment (nachomazzara) [inline contracts/oracles/ChainlinkOracle.sol:39]: Do we need to use safeMath here or we are ok? Can you add a comment about why we are not using safe match? maybe we can just check that `tolerance` is lower than `block.timestamp`

File: contracts/mocks/DummyDataFeed.sol (hunk 1 above)
Comment (fzavalia) [reply contracts/mocks/DummyDataFeed.sol]: This just describes literally how the value is used in this mock.
The updatedAt parameter is always returned as the block.timestamp.
The updatedAtOffset provides a way to test the tolerance.

**Pull request merged**